### ISSUE TITLE
[Feature Request] cblas_zomatadd should be supported.

### ISSUE THREAD

**Issue by xqch1983** [xianyi/OpenBLAS#4236]
The clbas_zomatadd should be supported same with the interface in MKL.  implement the addition of two matrixes as bellow:
C := alpha*op(A) + beta*op(B)

The API in MKL is [mkl_?omatadd](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-c/2023-1/mkl-omatadd.html), which scales and adds two matrices, as well as performing out-of-place transposition operations.

**Comment by martin-frbg** [comment]
So something like omatcopy plus axpby looping over all columns... may I ask what the use case for this is, apart from it being part of MKL's BLAS-like extensions ?

**Comment by brada4** [comment]
Cast matrices to vectors.
Scal one and axpy other?

**Comment by xqch1983** [comment]
> So something like omatcopy plus axpby looping over all columns... may I ask what the use case for this is, apart from it being part of MKL's BLAS-like extensions ?

Yes, I checked the use case of matrix is that rows=m, cols=1 or rows=1, cols=n, so this omatadd case can be replaced by omatcopy and axpy. Actually, users want to replace the MKL BLAS API using OpenBLAS API one-to-one directly.

**Comment by brada4** [comment]
So make supercompat header to do the casts. BLAS L1 is quite well optimised by modern compilers.

**Comment by martin-frbg** [comment]
> > So something like omatcopy plus axpby looping over all columns... may I ask what the use case for this is, apart from it being part of MKL's BLAS-like extensions ?
> 
> Yes, I checked the use case of matrix is that rows=m, cols=1 or rows=1, cols=n, so this omatadd case can be replaced by omatcopy and axpy. Actually, users want to replace the MKL BLAS API using OpenBLAS API one-to-one directly.

do you happen to have sample code and/or a testcase ? I'm primarily curious in what kind of context this particular function is useful (and of course it is a general problem with BLAS extensions that few libraries implement them - besides the one who "invented" it)
